### DISCUSSION TITLE
Integration with Github Actions and CI

### DISCUSSION THREAD

**Discussion by anandijain** [community/community#3455] [Codespaces]
Hello, I'm wondering if there are/could be tools to quickly run and rerun github actions in codespaces.
The problem is that in trying to debug CI, I make a single tweak, push, then wait 5-10 minutes for it to finish.
I imagine there is a way that the entire image doesn't need to be restarted, only refreshing what's needed.

I think it'd make dealing with actions a lot easier. I'm wondering what others have to say

**Comment by iloveitaly** [comment]
I'd love this as well

**Comment by thomas-sickert** [comment]
I'd also love to iterate more quickly on Actions. 

Unfortunately, as it stand today, codespaces run as a container and actions are given their own VM, so there's a bit of a mismatch in capability. I have heard good things about [act](https://github.com/nektos/act), which is designed to run actions "locally" which should work on Codespaces as well, but I haven't tried it myself. 

On the GitHub side, I opened a feature request to investigate the feasibility of doing something like this. 

Thanks for the discussion!

**Reply by goyalyashpal** [reply]
> [act](https://github.com/nektos/act), which is designed to run actions "locally"

that's nice



> On the GitHub side, I opened a feature request

link?
